pkg/api/v1alpha1/versionskew.go

```diff
@@ -0,0 +1,37 @@
+package v1alpha1
+
+import (
+	"fmt"
+	"math"
+
+	"k8s.io/apimachinery/pkg/util/version"
+)
+
+const supportedMinorVersionIncrement = 1
+
+// ValidateVersionSkew validates Kubernetes version skew between valid non-nil versions.
+func ValidateVersionSkew(oldVersion, newVersion *version.Version) error {
+	newVersionMajor := newVersion.Major()
+	newVersionMinor := newVersion.Minor()
+	oldVersionMajor := oldVersion.Major()
+	oldVersionMinor := oldVersion.Minor()
+
+	majorVersionDifference := math.Abs(float64(newVersionMajor) - float64(oldVersionMajor))
+
+	if majorVersionDifference > 0 {
```

Comment:
what about `< 0`

Reply:
Wouldn't this code be simplify if you just did `newVersion.LessThan(oldVersion)`?

Reply:
If we were to use `newVersion.LessThan(oldVersion)` we would not be able to separate out errors for major version decrement vs minor version decrement.